pkg/storages/s3/uploader.go

```diff
@@ -96,9 +96,11 @@ func CreateUploaderAPI(svc s3iface.S3API, partsize, concurrency int) s3managerif
 	return uploaderAPI
 }
 
-// TODO : unit tests
 func partitionStrings(strings []string, blockSize int) [][]string {
 	// I've unsuccessfully tried this with interface{} but there was too much of casting
+	if blockSize <= 0 {
+		return [][]string{}
```

Comment:
I guess in that just like in the case when blockSize exceeds the `len(strings)` we should return the `[]string{strings}`